packages/core/src/external/cda/process-attachments.ts

```diff
@@ -0,0 +1,259 @@
+} from "@medplum/fhirtypes";
+import { executeWithNetworkRetries } from "@metriport/shared";
+import { buildDayjs } from "@metriport/shared/common/date";
+import { MedicalDataSource, isMedicalDataSource } from "..";
+import { createAttachmentUploadFilePath } from "../../domain/document/upload";
+import {
+  CdaCodeCv,
+  CdaInstanceIdentifier,
+  CdaOriginalText,
+  ConcernActEntryAct,
+  EffectiveTimeLowHigh,
+} from "../../fhir-to-cda/cda-types/shared-types";
+import { executeAsynchronously } from "../../util/concurrency";
+import { Config } from "../../util/config";
+import { out } from "../../util/log";
+import { sizeInBytes } from "../../util/string";
+import { uuidv4 } from "../../util/uuid-v7";
+import { S3Utils, UploadParams } from "../aws/s3";
+import { cqExtension } from "../carequality/extension";
+import { cwExtension } from "../commonwell/extension";
+import { makeFhirApi } from "../fhir/api/api-factory";
+import { convertCollectionBundleToTransactionBundle } from "../fhir/bundle/convert-to-transaction-bundle";
+import { buildDocIdFhirExtension } from "../fhir/shared/extensions/doc-id-extension";
+
+const region = Config.getAWSRegion();
+
+function getS3UtilsInstance(): S3Utils {
+  return new S3Utils(region);
+}
+export async function processAttachments({
+  b64Attachments,
+  cxId,
+  patientId,
+  filePath,
+  medicalDataSource,
+  s3BucketName,
+  fhirUrl,
+}: {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  b64Attachments: any[];
+  cxId: string;
+  patientId: string;
+  filePath: string;
+  medicalDataSource?: string | undefined;
+  s3BucketName: string | undefined;
+  fhirUrl?: string;
+}) {
+  const { log } = out(`processAttachments - cxId ${cxId}, patientId ${patientId}`);
+  const s3Utils = getS3UtilsInstance();
+
+  const extensions = [buildDocIdFhirExtension(filePath), getSourceExtension(medicalDataSource)]
+    .flat()
+    .filter(Boolean) as Extension[];
+
+  const docRefs: DocumentReference[] = [];
+  const uploadDetails: UploadParams[] = [];
+
+  b64Attachments.map(async att => {
+    const act = att.act as ConcernActEntryAct;
+
+    const fileDetails = getDetails(act.text);
+    if (!fileDetails.fileB64Contents) return;
+
+    const docRef = buildDocumentReference(patientId, extensions, act);
+    if (!docRef.id) throw new Error("Missing ID in DocRef");
+
+    const fileKey = createAttachmentUploadFilePath({
+      filePath,
+      attachmentId: docRef.id,
+      mimeType: fileDetails.mimeType,
+    });
+
+    const fileUrl = s3Utils.buildFileUrl(s3BucketName ?? "", fileKey);
+
+    const attachment: Attachment = {
+      ...(fileDetails.mimeType && { contentType: fileDetails.mimeType }),
+      url: fileUrl,
+      size: sizeInBytes(fileDetails.fileB64Contents),
+      title: fileKey,
+    };
+
+    if (docRef.date) attachment.creation = docRef.date;
+    docRef.content = [{ attachment }];
+
+    docRefs.push(docRef);
+
+    const uploadParams = {
+      bucket: s3BucketName ?? "",
```

Comment:
only add this if the s3 bucket is defined -- we know this doesn't make sense otherwise.